tests/unit/challenges/test_views.py

```diff
@@ -443,17 +462,29 @@ def test_particular_challenge_for_disable_does_not_exist(self):
         self.url = reverse_lazy('challenges:disable_challenge',
                                 kwargs={'pk': self.challenge.pk + 2})
         response = self.client.post(self.url, {})
-        self.assertEqual(response.status_code, status.HTTP_406_NOT_ACCEPTABLE)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
 
     def test_when_user_does_not_have_permission_to_disable_particular_challenge(self):
         self.url = reverse_lazy('challenges:disable_challenge',
                                 kwargs={'pk': self.challenge2.pk})
         expected = {
-            'error': 'Sorry, you do not have permission to disable this challenge'
+            'detail': 'Sorry, you are not allowed to perform this operation!'
         }
         response = self.client.post(self.url, {})
         self.assertEqual(response.data, expected)
-        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_disable_challenge_when_user_is_not_creator(self):
+        self.url = reverse_lazy('challenges:disable_challenge',
+                                kwargs={'pk': self.challenge2.pk})
+        expected = {
+            'detail': 'Sorry, you are not allowed to perform this operation!'
+        }
+        alt_client = APIClient(enforce_csrf_checks=True)
```

Comment:
> If I understand this correctly, then I should add self.user as a ChallengeHost with the self.challenge_host_team1 team, so self.user now has access to the challenge but is not the creator. Does this sound correct to you?

Yep, this is correct. Just make sure that the user is not the creator of the team. Rest all sounds good.